src/main/java/com/gary/backendv2/service/TrustedPersonService.java

```diff
@@ -0,0 +1,114 @@
+package com.gary.backendv2.service;
+
+import com.gary.backendv2.exception.HttpException;
+import com.gary.backendv2.model.TrustedPerson;
+import com.gary.backendv2.model.User;
+import com.gary.backendv2.model.dto.request.AllergyRequest;
+import com.gary.backendv2.model.dto.request.TrustedPersonRequest;
+import com.gary.backendv2.model.dto.response.TrustedPersonResponse;
+import com.gary.backendv2.repository.TrustedPersonRepository;
+import com.gary.backendv2.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class TrustedPersonService {
+	private final TrustedPersonRepository trustedPersonRepository;
+	private final UserRepository userRepository;
+
+	public List<TrustedPersonResponse> getAll(){
+		List<TrustedPerson> trustedPeople = trustedPersonRepository.findAll();
+		List<TrustedPersonResponse> trustedPersonResponses = new ArrayList<>();
+		for (TrustedPerson t: trustedPeople) {
+			trustedPersonResponses.add(
+				TrustedPersonResponse
+						.builder()
+						.email(t.getEmail())
+						.firstName(t.getFirstName())
+						.trustedId(t.getTrustedId())
+						.lastName(t.getLastName())
+						.phone(t.getPhone())
+						.build()
+			);
+		}
+		return trustedPersonResponses;
+	}
+
+	public TrustedPersonResponse getByEmail(String userEmail){
+		Optional<User> userOptional = userRepository.findByEmail(userEmail);
+		if (userOptional.isEmpty()) {
+			throw new HttpException(HttpStatus.NOT_FOUND, String.format("Cannot find user with %s", userEmail));
+		}
+		TrustedPerson trustedPerson = userOptional.get().getTrustedPerson();
+		if(trustedPerson == null){
+			throw new HttpException(HttpStatus.NO_CONTENT, String.format("User %s don't have trusted person", userEmail));
+		}
+		return  TrustedPersonResponse
+				.builder()
+				.phone(trustedPerson.getPhone())
+				.email(trustedPerson.getEmail())
+				.lastName(trustedPerson.getLastName())
+				.firstName(trustedPerson.getFirstName())
+				.trustedId(trustedPerson.getTrustedId())
+				.build();
+	}
+
+	public void deleteByEmail(String userEmail){
+		Optional<User> userOptional = userRepository.findByEmail(userEmail);
+		if (userOptional.isEmpty()) {
+			throw new HttpException(HttpStatus.NOT_FOUND, String.format("Cannot find user with %s", userEmail));
+		}
+		TrustedPerson trustedPerson = userOptional.get().getTrustedPerson();
+		if(trustedPerson == null){
+			throw new HttpException(HttpStatus.NO_CONTENT, String.format("User %s don't have trusted person", userEmail));
```

Comment:
_The HTTP 204 No Content success status response code indicates that a request has succeeded, but that the client doesn't need to navigate away from its current page._

404 Not Found pls

Reply:
ewentualnie 200 OK ale bez response body

Reply:
10.2.5 204 No Content
The server has fulfilled the request but does not need to return an entity-body, and might want to return updated metainformation. The response MAY include new or updated metainformation in the form of entity-headers, which if present SHOULD be associated with the requested variant.

If the client is a user agent, it SHOULD NOT change its document view from that which caused the request to be sent. This response is primarily intended to allow input for actions to take place without causing a change to the user agent's active document view, although any new or updated metainformation SHOULD be applied to the document currently in the user agent's active view.

The 204 response MUST NOT include a message-body, and thus is always terminated by the first empty line after the header fields.

jednak 204 tu pasuje ale bez body, sam status HTTP as per RFC-2616

Reply:
no i w sumie pytanie czy chcemy być w 100% zgodni ze standardem http

Reply:
Nwm, przy x zapytaniach wysyłanych na raz fajnie jest mieć body które lepiej opisuje o co chodziło. Mogę zmienić na inny response taki który ma body by być zgodnym ze standardem - w ostateczności mogę dać not found

Reply:
zostaw jak jest w sumie